actors/market/src/lib.rs

```diff
@@ -459,11 +456,17 @@ impl Actor {
                     .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to set deal")?;
                 // Store verified allocation (if any) in the pending allocation IDs map.
                 // It will be removed when the deal is activated or expires.
-                if valid_deal.allocation != NO_ALLOCATION_ID {
+                if valid_deal.proposal.verified_deal {
                     msm.pending_deal_allocation_ids
                         .as_mut()
                         .unwrap()
-                        .set(deal_id_key(deal_id), valid_deal.allocation)
+                        .set(
+                            deal_id_key(deal_id),
+                            allocation_map
+                                .get_mut(&valid_deal.proposal.client.id().unwrap())
+                                .unwrap()
+                                .remove(0),
```

Comment:
This reliance on the matched ordering from construction of this map, and the mutability, is very subtle. Please instead construct a mapping from deal ID to allocation Id and look it up here. Maybe put the allocation ID back in ValidDeal

Reply:
In the latest modification, the transfer of datacap Is a batch operation outside of ValidDeal iter , it will return Vec<AllocationID>,What should i do to associate it with DealID